src/tui.rs

```diff
@@ -403,6 +409,18 @@ impl ToipeTui {
         //   - won't hang there waiting for user to type space
         lines.push(Text::from(line.join(" ")).with_faint());
 
+        if lines.len() + self.bottom_lines_len + 1 >= terminal_height.into() {
+            return Err(ToipeError::from(format!(
+                "toipe requires atleast {} lines in your terminal",
+                lines.len() + self.bottom_lines_len + 1
+            )));
+        } else if max_word_len >= max_width.into() {
+            return Err(ToipeError::from(format!(
```

Comment:
There is an issue with this. When the error occurs in between restarts, the errors gets printed first and then the screen gets cleared (by the `Drop` trait implementation of `ToipeTui`). So it appears as if toipe crashed without any message. Some terminals don't show the error even when scrolling back up (for me Alacritty showed the error as I scrolled up but Konsole did not).

I'm not sure how to work around this yet. It's okay if this can't be solved in this PR but would be nice to add a TODO somewhere, probably in the `impl Drop for ToipeTui`.

Reply:
good catch. even i'm not sure how to solve this yet. i'll open a new pr if i find a way to do this. adding a `TODO` comment, with a link to this discussion, as of now.